src/sentry/sentry_metrics/multiprocess.py

```diff
@@ -536,22 +538,92 @@ def __init__(
         self.__producer_topic = settings.KAFKA_TOPICS[settings.KAFKA_SNUBA_METRICS].get(
             "topic", "snuba-metrics"
         )
+        self.__commit_function = commit_function
 
         self.__closed = False
         self.__metrics = get_metrics()
+        self.__produced_message_offsets = {}
+        self.__callbacks = 0
+        self.__started = time.time()
+        # Need to make these flags
+        self.__commit_max_batch_size = 10000
+        self.__commit_max_batch_time = 4.0
+        self.__producer_queue_max_size = 80000
+        self.__producer_long_poll_timeout = 3.0
+
+        # poll duration metrics
+        self.__poll_start_time = time.time()
+        self.__poll_duration_sum = 0.0
+        self.__poll_sample_batch = 0
+
+    def _ready(self) -> bool:
+        if self.__callbacks >= self.__commit_max_batch_size:
+            return True
+        now = time.time()
+        if now >= (self.__started + self.__commit_max_batch_time):
+            return True
+
+        return False
+
+    def _record_poll_duration(self, poll_duration) -> None:
+        self.__poll_duration_sum += poll_duration
+
+        # record poll time durations every 5 seconds
+        if (self.__poll_start_time + 5) < time.time():
+            self.__metrics.incr(
+                "simple_produce_step.join_duration", amount=int(self.__poll_duration_sum)
```

Comment:
`simple_produce_step.join_duration` is a timer, you will have to record it as a timer here as well. The metrics object has a timing function `timing` to record timer metrics.